File: YGOJ_answer/1047/1047.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

string change(int n, int b) {
    string x = "";
    while (n > 0) {
        int c = n % b;
        x = to_string(c) + x;
        n = n / b;
    }
    return x;
}

bool pd(const string &a, const string &b) {
    char c = a[0], d = b[0];
    int i = 0, j = 0;
    if (c != d) {
        return false;
    }
    while (i < a.length() && j < b.length()) {
        if (a[i] == b[j]) {
            i++;
        }
        j++;
    }
    return i == a.length();
}

int main() {
    int m, n;
    cin >> m >> n;
    bool flag = true;
    for (int i = 2; i < 9; i++) {
        string m1 = change(m, i);
        string n1 = change(n, i);
        vector<int> list1(i, 0);
        vector<int> list2(i, 0);
        
        for (char k : m1) {
            list1[k - '0']++;
        }
        for (char k : n1) {
            list2[k - '0']++;
        }
        
        if (pd(m1, n1) && m1 != n1) {
            cout << i << " ";
            for (int j = 0; j < i; j++) {
                list2[j] -= list1[j];
            }
            for (int k = 0; k < list2.size(); k++) {
                if (list2[k] != 0) {
                    for (int l = 0; l < list2[k]; l++) {
                        cout << k;
                    }
                }
            }
            cout << endl;
            flag = false;
        }
    }
    if (flag) {
        cout << "404 not found" << endl;
    }
    return 0;
}

File: YGOJ_answer/1047/1047.py
'''m,n = input().split()
def tentoq(ten,q):
    ret=''
    while ten>0:
        r=ten%q
        if r<10:
            ret=chr(48+r)+ret
        else:
            ret=chr(55+r)+ret
        ten//=q
    return ret
jinzhi = []
for i in range(1,9):
    m1,n1 = m,n
    a = tentoq(n1,i)
    '''
def change(n,b):
    x=""
    while n>0:
        c=n%b
        x=str(c)+x
        n=n//b
    return x
def pd(a,b):
    c,d=a[0],b[0]
    i,j=0,0
    if c!=d:
        return False
    while i<len(a) and j<len(b):
        if a[i]==b[j]:
            i+=1
        j+=1
    if i==len(a):
        return True
    return False
m,n=map(int,input().split())
flag=True
for i in range(2,9):
    m1,n1=change(m,i),change(n,i)
    list1=[0]*i
    list2=[0]*i
    for k in m1:
        list1[int(k)]+=1
    for k in n1:
        list2[int(k)]+=1
    if pd(m1,n1) and m1!=n1:
        print(i,end=" ")
        for j in range(i):
            list2[j]-=list1[j]
        for k in range(len(list2)):
            if list2[k]!=0:
                for l in range(list2[k]):
                    print(str(k),end="")
        print()
        flag=False
if flag:
    print("404 not found")